packages/teams-js/src/public/media.ts

```diff
@@ -350,12 +367,125 @@ export namespace media {
    */
   export interface AudioProps {
     /**
-     * Optional; the maximum duration in minutes after which the recording should terminate automatically.
+     * Optional; the maximum duration in minutes after which the recording should terminate automatically
      * Default value is defined by the platform serving the API.
      */
     maxDuration?: number;
   }
 
+  /**
+   * @hidden
+   * Hide from docs
+   * --------
+   * Base class which holds the callback and notifies events to the host client
+   */
+  abstract class MediaController<T> {
+    protected controllerCallback: T;
+
+    public constructor(controllerCallback?: T) {
+      this.controllerCallback = controllerCallback;
+    }
+
+    protected abstract getMediaType(): MediaType;
+
+    /**
+     * @hidden
+     * Hide from docs
+     * --------
+     * This function will be implemented by the respective media class which holds the logic
+     * of specific events that needs to be notified to the app.
+     * @param mediaEvent indicates the event signed by the host client to the app
+     */
+    protected abstract notifyEventToApp(mediaEvent: MediaControllerEvent): void;
+
+    /**
+     * @hidden
+     * Hide from docs
+     * --------
+     * Function to notify the host client to programatically control the experience
+     * @param mediaEvent indicates what the event that needs to be signaled to the host client
+     */
+    protected notifyEventToHost(mediaEvent: MediaControllerEvent): Promise<void> {
+      return new Promise<void>(resolve => {
+        ensureInitialized(FrameContexts.content, FrameContexts.task);
+        const err = isApiSupportedOnMobile(nonFullScreenVideoModeAPISupportVersion);
+        if (err) {
+          throw err;
+        }
+
+        const params: MediaControllerParam = { mediaType: this.getMediaType(), mediaControllerEvent: mediaEvent };
+        resolve(sendAndHandleSdkError('media.controller', [params]));
+      });
+    }
+
+    /**
+     * Function to programatically stop the ongoing media event
+     */
+    public stop(): Promise<void> {
```

Comment:
Please make an overloaded function like 'notifyEventToHost'

Reply:
Also make it protected as the previous methods are protected, unless only stop method is to be accessed by outside classes. I would ask the implementer on what the use case for this.

Reply:
To me it makes sense that the stop() method is public since they're probably planning on adding more public functions like stop() that the client can call which call on the protected methods with specific arguments. I've found a link to the thread on this and messaged it to you, I think it's pretty helpful in understanding the direction of the added media functionalities. If it would still be recommended to use protected, I would love to pick your brain and find out more!

Reply:
This seems like a good candidate for a bug to follow up with the implementer.